perf_test/graph/KokkosGraph_mis_d2.cpp

```diff
@@ -0,0 +1,381 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Brian Kelley (bmkelle@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <stdlib.h>
+#include <string>
+#include <set>
+#include <unistd.h>
+
+#include <iostream>
+#include <iomanip>
+
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <ctime>
+#include <limits>
+#include <string>
+#include <sys/time.h>
+
+#include <Kokkos_Core.hpp>
+
+#include "KokkosKernels_Utils.hpp"
+#include "KokkosSparse_CrsMatrix.hpp"
+#include "KokkosSparse_spadd.hpp"
+#include "KokkosGraph_MIS2.hpp"
+#include "KokkosKernels_default_types.hpp"
+
+using namespace KokkosGraph;
+
+struct MIS2Parameters
+{
+  int repeat = 1;
+  bool verbose = false;
+  int use_threads = 0;
+  int use_openmp = 0;
+  int use_cuda = 0;
+  int use_serial = 0;
+  const char* mtx_file = NULL;
+  MIS2_Algorithm algo = MIS2_FAST;
```

Comment:
MIS2_QUALITY is not an option ?

Reply:
The table above is the MIS2_FAST algorithm. You're right, I should add MIS2_QUALITY as an option for the perf test, if I keep it at all (but it's way slower in exchange for not much increase in the mis size). Both are deterministic. I'll give some numbers from the graphs above to compare FAST against QUALITY.